tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/timeout/clientserver/UninterruptableTimeoutClient.java

```diff
@@ -94,7 +98,7 @@ public void serviceTimeout(long waitMs) {
                 Assert.fail("Waiting future threw exception", e);
             }
             catch (TimeoutException e) {
-                return CompletableFuture.completedFuture(null);
+                Assert.fail("Waiting future timed out", e);
```

Comment:
I wanted to ask if this is guaranteed to run on the testing thread (otherwise asserts are useless), but I see we do that already in the previous `catch` block.

Reply:
I did also test that this exception is thrown when it should be and that the failure message is thrown:

I configured a `WAITING_FUTURE_DURATION` less than the scaled `@Timeout value` and the failure message was returned as expected :)

Reply:
It doesn't run on the testing thread, it results in an `AssertionError` being thrown on the async thread, which is then returned to the testing thread via the `Future`. In almost all cases, the test is asserting the result from the `Future`.

I do spot one case where it isn't in `testTimeoutAsyncBulkheadQueueTimed`. @Joseph-Cass could you add a check to the result of the first call to `serviceTimeoutAsyncBulkheadQueueTimed`. Calling `future.get()` after completing `waitingFutureA` should be sufficient to ensure the failure exception is thrown.

Reply:
I see. Thanks!

Reply:
I have added an additional check for `testTimeoutAsyncBulkheadQueueTimed` in commit [a4555ff](https://github.com/eclipse/microprofile-fault-tolerance/pull/517/commits/a4555ff97991119719c3e57e629032dac09954fa), thanks for spotting the missing case @Azquelt